shm/Ship.hpp

```diff
@@ -30,11 +34,15 @@ class Ship {
     Ship& operator-=(size_t num);
     Ship& operator+=(size_t num);
 
+    void load(const std::shared_ptr<Cargo>& cargo) { allCargos_.push_back(cargo); }
+    //void unload(Cargo* cargo) {  }
```

Comment:
```suggestion
    void unload(Cargo* cargo) {
        if(!cargo->getAmount()){
            std::remove_if(cargo_.begin(), cargo_.end(), [cargo](const auto& ptr) {
                return ptr.get() == cargo;
            })
        }       
    }
```

Straight up unloading i, no amount subtraction etc but I think it's rather straightforward to add(compare by name and base price i think)

Reply:
Why ` if(!cargo->getAmount())` ? Why amount needs to be 0 ?

Reply:
It would delete the cargo from vector/ship only if we had no cargo. As this function is also supposed to be able to unload cargo partially. ie 20 out of 30 units. But that's not implemented.